app/models/resource_description_step.rb

```diff
@@ -19,8 +19,13 @@ def initialize(step = 'resource-description', title = "Resource description", su
 
   def execute context
     mediaobject = context[:mediaobject]
-    mediaobject.permalink = context[:media_object].delete(:permalink)
-    mediaobject.update_datastream(:descMetadata, context[:media_object])
+    populate_from_catalog = context[:media_object][:import_bib_record].present?
+    if populate_from_catalog
+      mediaobject.descMetadata.populate_from_catalog!(context[:media_object][:bibliographic_id])
+    else
+      mediaobject.permalink = context[:media_object].delete(:permalink)
```

Comment:
Why is the permalink not set if also populating from the catalog?  Could there be a case where you want to import from the catalog but also specify a new or different permalink?